### PR TITLE
bugfix: Fix Breadcrumb tests for iPhone [NP=1253]

### PR DESCRIPTION
**Pull request by celonicolosi**
Some breadcrumbs scenarios were falling on the iPhone because breadcrumb behaves differently on responsive screens.
This PR fixes those problems by expecting different results for mobile_phones.

### CODE REVIEW

File: testing/features/step_definitions/breadcrumbs_steps.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 Given("a Default Breadcrumbs component is on the page") do
   @component = Breadcrumbs::Default.new
   @component.load
+  @default = true
```
Comment (luke-hill) [inline testing/features/step_definitions/breadcrumbs_steps.rb:6]: This shouldn't be needed. We can either check the class, or do something else. But I'm fairly certain just the mobile_phone? check should be fine.

**Review by luke-hill — COMMENTED** [review]
Could we add a tag - `@small_screen` to the bottom scenario. And have it do 1 of 2 things inside our hooks.

1) If mobile, nothing
2) If desktop, resize to small res?

File: testing/features/step_definitions/breadcrumbs_steps.rb (hunk 1 above)
Comment (celonicolosi) [reply testing/features/step_definitions/breadcrumbs_steps.rb:6]: The problem here is not the small_screen of mobile_phone. The problem is in the step "But the final breadcrumb isn't a link", that is not true for breadcrumb_default:
Screenshot 2020-12-16 at 17 38 21
So maybe we need to change the step to be like: But the final breadcrumb isn't a link unless on default? What do you think @luke-hill

**Comment by luke-hill** [comment]
If the specs i.e. the physical specification is different on different variants, then that's fine. Reflect that in the tests. But yeh we should still test all variants where they are functionally possible (So instead of mobile only, make ti small screen only)

File: testing/features/breadcrumbs.feature [hunk 2]
```diff
@@ -8,6 +7,15 @@ Feature: Breadcrumbs component
     Background:
       Given a Default Breadcrumbs component is on the page
 
+    Scenario: Breadcrumbs only link up till the last one
+      Then a series of breadcrumbs are present
+      And the initial breadcrumbs are all links
+      But the final breadcrumb isn't a link unless on mobile
```
Comment (luke-hill) [inline testing/features/breadcrumbs.feature:13]: lets remove this "unless on mobile" and just not assert. The previous cuke will validate all breadcrumbs are links.

File: testing/features/step_definitions/breadcrumbs_steps.rb [hunk 3]
```diff
@@ -10,14 +10,31 @@
   @component.load
 end
 
+Given("a Site Wide Breadcrumbs component is on the page") do
+  @component = Breadcrumbs::SiteWide.new
```
Comment (luke-hill) [inline testing/features/step_definitions/breadcrumbs_steps.rb:14]: can we tap load this and the other, to save space.

**Review by luke-hill — APPROVED** [review]
@davidrapson Are you ok with this too. I think this distinction is quite good (As we start using the concept of small screens, not just it's a mobile)

**Comment by davidrapson** [comment]
Yep, definitely prefer thinking in terms of breakpoints, not devices (unless it's genuinely device specific)

**Pull request merged**